cuegui/cuegui/AbstractGraphWidget.py

```diff
@@ -0,0 +1,187 @@
+from PySide2 import QtCore
+from PySide2 import QtGui
+from PySide2 import QtWidgets
+
+from NodeGraphQt import NodeGraph
+from cuegui.CueNodeGraphQt import CueLayerNode
+
+
+class AbstractGraphWidget(QtWidgets.QWidget):
+
+    def __init__(self, parent=None):
+        super(AbstractGraphWidget, self).__init__(parent=parent)
+        self.setupUI()
+
+        self.timer = QtCore.QTimer(self)
+        self.timer.timeout.connect(self.update)
+        self.timer.setInterval(1000 * 20)
+
+        self.graph.node_selection_changed.connect(self.on_node_selection_changed)
+        QtGui.qApp.quit.connect(self.timer.stop)
+
+    def setupUI(self):
+        '''Setup the UI.'''
+        self.graph = NodeGraph()
+        try:
+            self.graph.register_node(CueLayerNode)
+            self.graph.register_node(BackdropNode)
+        except Exception:
+            pass
+        self.graph.viewer().installEventFilter(self)
+
+        # disable editing node connections/dropping new nodes
+        self.graph.viewer().search_triggered.disconnect(
+            self.graph._on_search_triggered
+        )
+        self.graph.viewer().connection_sliced.disconnect(
+            self.graph._on_connection_sliced
+        )
+        self.graph.viewer().connection_changed.disconnect(
+            self.graph._on_connection_changed
+        )
+
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.addWidget(self.graph.viewer())
+
+    def on_node_selection_changed(self):
+        '''Slot run when a node is selected.
+
+        Updates the nodes to ensure they're visualising current data.
+        Can be used to notify other widgets of object selection.
+        '''
+        self.update()
+
+    def handle_select_objects(self, rpcObjects):
+        '''Select incoming objects in graph.
+        '''
+        received = [o.data.name for o in rpcObjects]
+        current = [rpcObject.data.name for rpcObject in self.selected_objects()]
```

Comment:
rpcObjects job, layer, frame have the api call `name()`, which is encouraged to use over the native `.data.name` call